test/test-pexprs-madlibs.js

```diff
@@ -0,0 +1,195 @@
+'use strict';
+
+var test = require('tape');
+
+var makeGrammar = require('./testUtil').makeGrammar;
+
+// --------------------------------------------------------------------
+// Tests
+// --------------------------------------------------------------------
+
+// test('PExpr');
+//
+test('Alt', function(t) {
+  var g = makeGrammar([
+    'G {',
+    '  alt = "a" | "b"',
+    '}'
+  ]);
+
+  var random = Math.random;
+
+  Math.random = function() { return 0.6; };
+  var example = g.rules.alt.body.generateExample(g, {}, false, []);
+  t.equal(example.example, 'b');
+
+  Math.random = function() { return 0.2; };
+  example = g.rules.alt.body.generateExample(g, {}, false, []);
+  t.equal(example.example, 'a');
+
+  Math.random = random;
+  t.end();
+});
+
+test('Seq', function(t) {
+  var g = makeGrammar([
+    'G {',
+    '  seq = "a" "b"',
+    '}'
+  ]);
+
+  var example = g.rules.seq.body.generateExample(g, {}, false, []);
+
+  t.equal(example.example, 'ab');
+
+  t.end();
+});
+
+test('Apply', function(t) {
+  var g = makeGrammar([
+    'G {',
+    '  apply = a ',
+    '  a = "b"*   ',
+    '}'
+  ]);
+
+  var example = g.rules.apply.body.generateExample(g, {a: ['bbbb']}, false, []);
+
+  t.equal(example.example, 'bbbb');
+  t.end();
+});
+
+test('Terminal', function(t) {
+  var g = makeGrammar([
+    'G {',
+    '  a = "a"',
+    '}'
+  ]);
+
+  var example = g.rules.a.body.generateExample(g, {}, false, []);
+
+  t.equal(example.example, 'a');
+  t.end();
+});
+
+test('Range', function(t) {
+  var g = makeGrammar([
+    'G {',
+    '  num = "1" .. "9"',
+    '}'
+  ]);
+
+  var example = g.rules.num.body.generateExample(g, {}, false, []);
+  var charCode = example.example.charCodeAt(0);
+
+  t.pass(charCode >= '1'.charCodeAt(0));
```

Comment:
This is probably not doing what you intended...see https://github.com/substack/tape#tpassmsg. It should probably be `t.ok(...)` or `t.equal(..., true)`.